pkg/ndctl/region.go

```diff
@@ -208,13 +204,36 @@ func (r *region) CreateNamespace(opts CreateNamespaceOpts) (Namespace, error) {
 	}
 
 	if opts.Size != 0 {
-		ways := uint64(C.ndctl_region_get_interleave_ways(r))
-		align = align * ways
-		if opts.Size%align != 0 {
-			// Round up size to align with next block boundary.
-			opts.Size = (opts.Size/align + 1) * align
-			klog.V(4).Infof("%s: namespace size must align to interleave-width:%d * alignment:%d, force-align to %d",
-				regionName, ways, align, opts.Size)
+		/* old method, using namespace align and interleave ways.
+		If we use fixed 2M namespace align, this is always overshadowed
+		by region align which is 16 or 96 MB so we may leave this code out
+		ways := r.InterleaveWays()
+				align = align * ways
+				if opts.Size%align != 0 {
+					// Align down to namespace-alignment boundary.
+					opts.Size /= align
+					opts.Size *= align
+					klog.V(4).Infof("%s: namespace size must align to interleave-width:%d * alignment:%d, adjust to %d",
+						regionName, ways, align, opts.Size)
+				}*/
```

Comment:
Please remove the old code and instead add comments explaining how the current code works.